src/main/java/org/mifos/connector/mpesa/flowcomponents/state/TransactionStateWorker.java

```diff
@@ -0,0 +1,80 @@
+package org.mifos.connector.mpesa.flowcomponents.state;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.zeebe.client.ZeebeClient;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.support.DefaultExchange;
+import org.mifos.connector.common.gsma.dto.GSMATransaction;
+import org.mifos.connector.mpesa.dto.TransactionStatusRequestDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.Map;
+
+import static org.mifos.connector.mpesa.camel.config.CamelProperties.*;
+import static org.mifos.connector.mpesa.zeebe.ZeebeVariables.TRANSFER_RETRY_COUNT;
+
+@Component
+public class TransactionStateWorker {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private ProducerTemplate producerTemplate;
+
+    @Autowired
+    private CamelContext camelContext;
+
+    @Autowired
+    private ZeebeClient zeebeClient;
+
+    @Value("${zeebe.client.evenly-allocated-max-jobs}")
+    private int workerMaxJobs;
+
+    @PostConstruct
+    public void setupWorkers() {
+
+        zeebeClient.newWorker()
+                .jobType("transactionState")
+                .handler((client, job) -> {
+                    logger.info("Job '{}' started from process '{}' with key {}", job.getType(), job.getBpmnProcessId(), job.getKey());
+                    Map<String, Object> variables = job.getVariablesAsMap();
+                    variables.put(TRANSFER_RETRY_COUNT, 1 + (Integer) variables.getOrDefault(TRANSFER_RETRY_COUNT, 0));
+
+                    // TODO HOW TO GET [TransactionStatusRequestDTO] OBJECT?
```

Comment:
for `direct:lipana-transaction-status` to work it needs `TransactionStatusRequestDTO` object how can I get that? Since this is system generated request and not channel connector request so I am sure how to process the worker.

Reply:
Why do you need it in the worker?

You won't need to do a 1:1 copy of GSMA. 
I need to do it in the GSMA cause of requestingLei and once a variable is introduced in the workflow it stays available throughout across all the workers.
```
GSMATransaction gsmaChannelRequest = objectMapper.readValue((String) variables.get("gsmaChannelRequest"), GSMATransaction.class);
exchange.setProperty(RECEIVING_TENANT, gsmaChannelRequest.getReceivingLei());
```

Reply:
I need it in the worker because, the worker is the starting point of the camel flow.... so its the only place which will be calling the camel rout `direct:lipana-transaction-status`